scripts/check_copyright_header/src/main.rs

```diff
@@ -0,0 +1,167 @@
+            "".to_string(),
+            "// Rest of the file...".to_string(),
+        ]
+        .into_iter()
+        .map(Result::Ok);
+
+        assert!(check_file_header(lines).is_ok());
+    }
+
+    #[test]
+    fn test_valid_file_with_multiple_headers() {
+        let lines = vec![
+            "// Copyright (c) Zefchain Labs, Inc.".to_string(),
+            "// Copyright (c) Some Other Company".to_string(),
+            "// SPDX-License-Identifier: Apache-2.0".to_string(),
+            "".to_string(),
+            "// Rest of the file...".to_string(),
+        ]
+        .into_iter()
+        .map(Result::Ok);
+
+        assert!(check_file_header(lines).is_ok());
+    }
+
+    #[test]
+    fn test_invalid_file_missing_zefchain_header() {
+        let lines = vec![
+            "// SPDX-License-Identifier: Apache-2.0".to_string(),
+            "".to_string(),
+            "// Rest of the file...".to_string(),
+        ]
+        .into_iter()
+        .map(Result::Ok);
+
+        assert_eq!(
+            check_file_header(lines)
+                .map_err(|err| err.to_string())
+                .unwrap_err(),
+            "Incorrect copyright header, Zefchain Labs header not found"
+        );
+    }
+
+    #[test]
+    fn test_invalid_file_incorrect_zefchain_header() {
+        let lines = vec![
+            "// Copyright (c) Some Other Company".to_string(),
+            "// SPDX-License-Identifier: Apache-2.0".to_string(),
+            "".to_string(),
+            "// Rest of the file...".to_string(),
+        ]
+        .into_iter()
+        .map(Result::Ok);
+
+        assert_eq!(
+            check_file_header(lines)
+                .map_err(|err| err.to_string())
+                .unwrap_err(),
+            "Incorrect copyright header, Zefchain Labs header not found"
+        );
+    }
+
+    #[test]
+    fn test_invalid_file_unexpected_line() {
+        let lines = vec![
+            "// Copyright (c) Zefchain Labs, Inc.".to_string(),
+            "// SPDX-License-Identifier: Apache-2.0".to_string(),
+            "Unexpected line".to_string(),
+            "".to_string(),
+            "// Rest of the file...".to_string(),
+        ]
+        .into_iter()
+        .map(Result::Ok);
+
+        assert_eq!(
+            check_file_header(lines)
+                .map_err(|err| err.to_string())
+                .unwrap_err(),
+            "Separation line not found"
+        );
+    }
+
+    #[test]
+    fn test_invalid_file_empty_line_before_header() {
+        let lines = vec![
+            "".to_string(),
+            "// SPDX-License-Identifier: Apache-2.0".to_string(),
+            "".to_string(),
+            "// Rest of the file...".to_string(),
+        ]
+        .into_iter()
+        .map(Result::Ok);
+
+        assert_eq!(
+            check_file_header(lines)
+                .map_err(|err| err.to_string())
+                .unwrap_err(),
+            "Unexpected line reached"
```

Comment:
These asserts on the error make me realize that `anyhow` was probably not as good as `thiserror` in this case. We can convert another time (although it would be a great exercise to learn `thiserror`).

Reply:
Oh ok, I'll do it in a follow up PR